src/main/java/com/conveyal/datatools/manager/controllers/api/FeedSourceController.java

```diff
@@ -277,9 +278,12 @@ public static FeedSource requestFeedSourceById(Request req, String action) {
         return requestFeedSource(req, FeedSource.get(id), action);
     }
     public static FeedSource requestFeedSource(Request req, FeedSource s, String action) {
-        Auth0UserProfile userProfile = req.attribute("user");
         Boolean publicFilter = Boolean.valueOf(req.queryParams("public")) ||
-                req.url().split("/api/*/")[1].startsWith("public");
+                req.url().split("/api/manager/*/")[1].startsWith("public");
+        if (!publicFilter) {
```

Comment:
Why having a guard clause and a ternary operator checking against the same scenario ? Alternative proposition :

```java
if(publicFilter) {
  Auth0UserProfile userProfile = null;
} else {
  Auth0Connection.checkUser(req);
  Auth0UserProfile userProfile = req.attribute("user");
}
```